platform/default/mbgl/storage/offline_download.cpp

```diff
@@ -263,4 +263,18 @@ void OfflineDownload::ensureResource(const Resource& resource, std::function<voi
     });
 }
 
+bool OfflineDownload::hasExceededTileCount(const Resource& resource) {
+    bool tileCountExceeded = false;
+    
+    if (resource.kind == Resource::Kind::Tile
+        && util::mapbox::isMapboxURL(resource.url)
+        && offlineDatabase.offlineMapboxTileCountLimitExceeded()) {
+        observer->mapboxTileCountLimitExceeded(offlineDatabase.getOfflineMapboxTileCountLimit());
+        setState(OfflineRegionDownloadState::Inactive);
+        tileCountExceeded = true;
```

Comment:
MBGL core convention is to prefer not to use temporary variables if possible, so use `return true;` here and `return false;` down below, and remove the `tileCountExceeded` variable.